### ISSUE TITLE
Broken on MinGW due to psutil dependency

### ISSUE THREAD

**Issue by mmha** [conan-io/conan#2058]
26ea5f0 introduced a dependency on `psutil`. Unfortunately mainline `psutil` does not support MinGW and/or MSYS (There was a PR at some point: giampaolo/psutil#845).

**Comment by memsharded** [comment]
Hi @mmha 

thanks for reporting this issue! Could you please clarify what is the error? Does conan upgrade well, then it fails when building with MinGW some package? It should be unrelated, those are different parts of the codebase, doing different things, I am a bit confused.

**Comment by mmha** [comment]
I tried to install conan on a fresh MSYS2 installation. I didn't try to update my other system yet, but I assume it would fail with the same error:

```
pip install conan
Collecting conan
  Using cached conan-0.29.0.tar.gz
Collecting PyJWT<2.0.0,>=1.4.0 (from conan)
  Using cached PyJWT-1.5.3-py2.py3-none-any.whl
Collecting requests<3.0.0,>=2.7.0 (from conan)
  Using cached requests-2.18.4-py2.py3-none-any.whl
Collecting colorama<0.4.0,>=0.3.3 (from conan)
  Using cached colorama-0.3.9-py2.py3-none-any.whl
Collecting PyYAML<3.13.0,>=3.11 (from conan)
  Using cached PyYAML-3.12.tar.gz
Collecting patch==1.16 (from conan)
  Using cached patch-1.16.zip
Collecting fasteners>=0.14.1 (from conan)
  Using cached fasteners-0.14.1-py2.py3-none-any.whl
Requirement already satisfied: six>=1.10.0 in /usr/lib/python3.6/site-packages (from conan)
Collecting node-semver==0.2.0 (from conan)
  Using cached node_semver-0.2.0-py3-none-any.whl
Collecting distro<1.1.0,>=1.0.2 (from conan)
  Using cached distro-1.0.4-py2.py3-none-any.whl
Collecting pylint<=1.8.0,>=1.6.5 (from conan)
  Using cached pylint-1.7.4-py2.py3-none-any.whl
Collecting future==0.16.0 (from conan)
  Using cached future-0.16.0.tar.gz
Collecting pygments<3.0,>=2.0 (from conan)
  Using cached Pygments-2.2.0-py2.py3-none-any.whl
Collecting psutil>=5.4.1 (from conan)
  Using cached psutil-5.4.1.tar.gz
    Complete output from command python setup.py egg_info:
    platform msys is not supported

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-qsmr8ya_/psutil/
```

There is also a patched version of psutil available in the MSYS2 repository (`mingw32/mingw-w64-i686-python2-psutil` and `mingw64/mingw-w64-x86_64-python2-psutil`), but they are outdated (Version 5.2.2).

So a workaround might be to just require `psutil>=5.2.2` if that's recent enough.

**Comment by memsharded** [comment]
I am going to try ``psutil==5.2.2``

**Comment by memsharded** [comment]
So, if we define the requirement as ``psutil>=5.2.2``, it will isntall 5.2.2 in MSYS2, because it is found in the MSYS2 repo?

**Comment by mmha** [comment]
Pip won't try to install the mainline `psutil` from the pypa repository, because it already found the patched `psutil 5.2.2` installed via the MSYS2 repository:

```
pacman -S mingw-w64-x86_64-python3 mingw-w64-x86_64-pip mingw-w64-x86_64-python3-psutil 
[Switch to the MINGW64 shell]
pip3 install -e . # in the conan directory with a downgraded psutil requirement
[...]
Requirement already satisfied: psutil>=5.2.2 in c:/msys64/mingw64/lib/python3.6/site-packages (from conan==0.29.0)
```
This is just a workaround and relies on the patch for an outdated version for `psutil` I linked in my original post. It's best to either drop the `psutil` dependency somehow or help them merge the PR that reintroduces MinGW support.

**Comment by memsharded** [comment]
Thanks very much for the feedback!

> It's best to either drop the psutil dependency somehow

That would require duplicating (probably with more bugs) the functionality provided by psutil

>help them merge the PR that reintroduces MinGW support

But that PR has been there for quite a long, there are no guarantees that it will be fixed and released soon, and we should try to provide a solution the sooner the better.


So, if it will work for MSYS2 if we specify ``psutil>=5.2.2``, then I think we should go for it. Following the other two approaches will badly affect most of conan users not suffering this MSYS2/psutil limitation. Then, try to follow up that open PR.

**Comment by memsharded** [comment]
We are going to remove psutil in today minor 0.29.1 release

**Comment by memsharded** [comment]
Released in conan 0.29.1
